packages/session-replay-browser/src/session-replay.ts

```diff
@@ -347,9 +323,10 @@ export class SessionReplay implements AmplitudeSessionReplay {
 
     this.recordCancelCallback = record({
       emit: (event) => {
-        const globalScope = getGlobalScope();
-        if ((globalScope && globalScope.document && !globalScope.document.hasFocus()) || !this.getShouldRecord()) {
-          this.stopRecordingAndSendEvents();
+        if (this.shouldOptOut()) {
```

Comment:
Should we have been doing this before? I wasn't sure if there was part of this refactor that caused us to add this now? Do we expect the opt out to happen during the middle of a session so that we need to check this on every emit? It seems like this is hitting some store that may affect performance?

Reply:
We actually were doing this before, a little more implicitly. We check for shouldOptOut within `getShouldRecord`, so the previous code path would've done the same behavior as this does. This is likely not how customers would use our SDK, but I added this logic just in case they want to start recording anonymous users, then stop recording them once they've done something like clicked 'don't track me' on a cookie acceptance modal. I think it's more likely that customers would only start recording after a user accepts cookies to be on the safer side, but that's the background on this anyways. I do think we could leave it out if there is concern about the logic here - as far as performance, I don't have a concern there, the identity store we look for the optOut value in is just an in memory store: https://github.com/amplitude/experiment-js-client/blob/main/packages/analytics-connector/src/identityStore.ts